js/liquid.js

```diff
@@ -434,37 +418,76 @@ module.exports = class liquid extends Exchange {
+            const margin = this.safeValue (market, 'margin_enabled');
+            const symbol = base + '/' + quote;
+            const maker = this.fees['trading']['maker'];
+            const taker = this.fees['trading']['taker'];
+            const parsedMarket = {
                 'id': id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
+                'settle': undefined,
                 'baseId': baseId,
                 'quoteId': quoteId,
+                'settleId': undefined,
                 'type': type,
                 'spot': spot,
+                'margin': spot && margin,
                 'swap': swap,
-                'maker': maker,
+                'future': false,
+                'option': false,
+                'active': !disabled,
+                'contract': swap,
+                'linear': undefined,
+                'inverse': undefined,
                 'taker': taker,
-                'limits': limits,
-                'precision': precision,
-                'active': active,
+                'maker': maker,
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': {
+                    'amount': this.parsePrecision ('8'),
+                    'price': this.safeNumber (market, 'tick_size'),
+                },
+                'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'amount': {
+                        'min': minAmount,
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': minPrice,
+                        'max': maxPrice,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
                 'info': market,
-            });
+            };
+            if (swap) {
+                const settlementCurrencies = this.options['fetchMarkets']['settlementCurrencies'];
+                for (let i = 0; i < settlementCurrencies.length; i++) {
+                    const settle = settlementCurrencies[i];
+                    parsedMarket['settle'] = settle;
+                    parsedMarket['symbol'] = symbol + ':' + settle;
+                    parsedMarket['linear'] = quote === settle;
+                    parsedMarket['inverse'] = base === settle;
+                    parsedMarket['taker'] = this.safeNumber (market, 'taker_fee', taker);
+                    parsedMarket['maker'] = this.safeNumber (market, 'maker_fee', maker);
+                    parsedMarket['contractSize'] = this.parseNumber ('1');
+                    result.push (parsedMarket);
```

Comment:
There's multiple funding currencies, so I thought this might be the best way to deal with that

![](https://camo.githubusercontent.com/8e167aea9e14d8138bc51f6b233acc20c84a8125aea2627d8beb406abf319eed/68747470733a2f2f692e696d6775722e636f6d2f33784c3166316f2e706e67)

Reply:
@kroitor So was this the right way to do this? And should I do something similar for bitfinex2?